File: remote_switch_v2/delphi/RemoteSwitchV2.pas
program RemoteSwitchV2;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletIndustrialQuadRelayV2;

type
  TRemoteSwitchV2 = class
  private
    ipcon: TIPConnection;
    iqr: TBrickletIndustrialQuadRelayV2;
  public
    procedure Execute;
    procedure AOn;
    procedure AOff;
    procedure BOn;
    procedure BOff;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '2g2gRs'; { Change to your UID }

var
  rs: TRemoteSwitchV2;

procedure TRemoteSwitchV2.AOn;
begin
  { Close channels 0 and 2 for 1.5 seconds }
  iqr.SetMonoflop(0, true, 1500);
  iqr.SetMonoflop(2, true, 1500);
end;

procedure TRemoteSwitchV2.AOff;
begin
  { Close channels 0 and 3 for 1.5 seconds }
  iqr.SetMonoflop(0, true, 1500);
  iqr.SetMonoflop(3, true, 1500);
end;

procedure TRemoteSwitchV2.BOn;
begin
  { Close channels 1 and 2 for 1.5 seconds }
  iqr.SetMonoflop(1, true, 1500);
  iqr.SetMonoflop(2, true, 1500);
end;

procedure TRemoteSwitchV2.BOff;
begin
  { Close channels 1 and 3 for 1.5 seconds }
  iqr.SetMonoflop(1, true, 1500);
  iqr.SetMonoflop(3, true, 1500);
end;

procedure TRemoteSwitchV2.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  iqr := TBrickletIndustrialQuadRelayV2.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  AOn();
end;

begin
  rs := TRemoteSwitchV2.Create;
  rs.Execute;
  rs.Destroy;
end.

File: remote_switch_v2/vbnet/RemoteSwitchV2.vb
Imports Tinkerforge

Module RemoteSwitchV2
	Const HOST As String = "localhost"
	Const PORT As Integer = 4223
	Const UID As String = "2g2gRs" ' Change to your UID

	Sub AOn(ByVal iqr As BrickletIndustrialQuadRelayV2)
		' Close channels 0 and 2 for 1.5 seconds
		iqr.SetMonoflop(0, true, 1500)
		iqr.SetMonoflop(2, true, 1500)
	End Sub

	Sub AOff(ByVal iqr As BrickletIndustrialQuadRelayV2)
		' Close channels 0 and 3 for 1.5 seconds
		iqr.SetMonoflop(0, true, 1500)
		iqr.SetMonoflop(3, true, 1500)
	End Sub

	Sub BOn(ByVal iqr As BrickletIndustrialQuadRelayV2)
		' Close channels 1 and 2 for 1.5 seconds
		iqr.SetMonoflop(1, true, 1500)
		iqr.SetMonoflop(2, true, 1500)
	End Sub

	Sub BOff(ByVal iqr As BrickletIndustrialQuadRelayV2)
		' Close channels 1 and 3 for 1.5 seconds
		iqr.SetMonoflop(1, true, 1500)
		iqr.SetMonoflop(3, true, 1500)
	End Sub

	Sub Main()
		Dim ipcon As New IPConnection() ' Create IP connection
		Dim iqr As New BrickletIndustrialQuadRelayV2(UID, ipcon) ' Create device object

		ipcon.Connect(HOST, PORT) ' Connect to brickd
		' Don't use device before ipcon is connected

		Aon(iqr)

		ipcon.Disconnect()
	End Sub
End Module
